Wangscape/Exp.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+#include <noise/noise.h>
+#include <cmath>
+
+// This is similar to noise::module::Exp.
+// However, no rescaling is performed,
+// and no attempt is made to guard against invalid operations.
+// Make sure the range of the source module
+// is compatible with the base of the exponential!
+// This module names its parameter "base" rather than "exponent",
+// because the formula is pow(base, source_value),
+// not pow(source_value, exponent).
+class Exp : public noise::module::Module
+{
+public:
+    double base;
+    Exp() :
+        Module(GetSourceModuleCount()),
+        mBase(2.)
+    {};
+    virtual int GetSourceModuleCount() const
+    {
+        return 1;
```

Comment:
This is a quirk of the libnoise library. It uses this number to allocate an array of `noise::Module*`s used as input values. For instance, a `Checkerboard` module has 0 inputs, an `Abs` module has 1 input, an `Add` module has 2.

We do need to specify this number manually for each class derived from `Module`, but we could add some intermediate abstract classes with 0, 1, 2... input modules. Then the custom `Module`s could be derived from those with less verbosity.